tests/client/bulk.rs

```diff
@@ -71,3 +71,103 @@ fn bulk_unordered_insert_only() {
         }
     }
 }
+
+#[test]
+fn bulk_ordered_mix() {
+    let models = vec![
+        WriteModel::InsertOne { document: doc! {
+            "_id" => (1),
+            "x" => (11)
+        }},
+        WriteModel::InsertOne { document: doc! {
+            "_id" => (2),
+            "x" => (22)
+        }},
+        WriteModel::InsertOne { document: doc! {
+            "_id" => (3),
+            "x" => (33)
+        }},
+        WriteModel::InsertOne { document: doc! {
+            "_id" => (4),
+            "x" => (44)
+        }},
+        WriteModel::ReplaceOne {
+            filter: doc! { "_id" => (3) },
+            replacement: doc! { "x" => (37) },
+            upsert: true,
+        },
+        WriteModel::UpdateMany {
+            filter: doc! { "_id" => { "$lt" => (3) } },
+            update: doc! { "$inc" => { "x" => (1) } },
+            upsert: false,
+        },
+        WriteModel::DeleteOne { filter: doc! {
+            "_id" => (4)
+        }},
+        WriteModel::InsertOne { document: doc! {
+            "_id" => (5),
+            "x" => (55)
+        }},
+        WriteModel::UpdateOne {
+            filter: doc! { "_id" => (6) },
+            update: doc! { "$set" =>  { "x" => (62) } },
+            upsert: true
+        },
+        WriteModel::InsertOne { document: doc! {
+            "_id" => (101),
+            "x" => ("dalmations")
+        }},
+        WriteModel::InsertOne { document: doc! {
+            "_id" => (102),
+            "x" => ("strawberries")
+        }},
+        WriteModel::InsertOne { document: doc! {
+            "_id" => (103),
+            "x" => ("blueberries")
+        }},
+        WriteModel::InsertOne { document: doc! {
+            "_id" => (104),
+            "x" => ("bananas")
+        }},
+        WriteModel::DeleteMany { filter: doc! {
+            "_id" => { "$gte" => (103) }
+        }},
+    ];
+
+    let client = MongoClient::new("localhost", 27017).unwrap();
+    let db = client.db("test");
+    let coll = db.collection("bulk_ordered_mix");
+
+    coll.drop().unwrap();
+
+    let result = coll.bulk_write(models, true);
+
+    assert_eq!(result.inserted_count, 9);
+    assert_eq!(result.inserted_ids.len() as i32, result.inserted_count);
+    assert_eq!(result.matched_count, 3);
+    assert_eq!(result.modified_count, 3);
+    assert_eq!(result.deleted_count, 3);
+    assert_eq!(result.upserted_count, 1);
+    assert_eq!(result.upserted_ids.len() as i32, result.upserted_count);
+
+    macro_rules! check_value_in_tree {
```

Comment:
What exactly do you want changed here? Just the macro name? It seems rather inconsequential, given that it's local to the function and therefore won't be used elsewhere.